### DISCUSSION TITLE
URL-based searching

### DISCUSSION THREAD

**Discussion by taliamath** [microsoft-search/pnp-modern-search#2257] [Q&A]
I want to create links to searches in PnP Modern Search. The syntax used for the traditional Sharepoint search does not work; what is the pattern for creating an https link that incorporates search terms?

**Comment by FranckyC** [comment]
You can use any URL query string parameter you want or URL fragment:

![image](https://user-images.githubusercontent.com/5659122/181346339-4a010b02-ecdf-4cc5-9dbc-48c0caaef3bd.png)

For query string parameters, they need to be already in the URL when you configure the connection, so you can see them as options.

**Comment by taliamath** [comment]
Changing the Source to "Page environment" disconnects the search box... is
there a way to use either the search box OR the page environment? Can you
give me an example of how I would construct the URL to pass the query to
the search results page (which in my case is the same page)?

**Reply by FranckyC** [reply]
Do the configuration in the search box WP like this:

![image](https://user-images.githubusercontent.com/5659122/181352441-7599878a-28d5-4e44-8b81-12e9540331b2.png)

And then connect the search box to the search results

**Reply by taliamath** [reply]
This does not really solve my issue with constructing the URL. For example, in traditional Sharepoint search, I have a search results page which takes keywords entered into the search box and turns them into an URL; for example:

https://mysite.sharepoint.com/search/Pages/results.aspx?k=words

With the PnP search, I have a URL for the page like this:

https://mysite.sharepoint.com/site/SitePages/test.aspx

When I try this URL: https://mysite.sharepoint.com/site/SitePages/test.aspx?k=words, it does not work as a search. It is ignored, and no search is performed.

**Reply by FranckyC** [reply]
@taliamath did you configure the search box to use the **k** parameter and connect to the search box to search results? If you want to set the query string parameter from the search box, you can configure the search box like this:

![image](https://user-images.githubusercontent.com/5659122/181355318-91a75f47-e800-452e-8f3e-c6eb1be2d701.png)

**Reply by taliamath** [reply]
@FranckyC Thanks, I will try this. I didn't realize these options were present, because I was not sending the results to a new page, so the options were hidden.

**Reply by taliamath** [reply]
@FranckyC I did what you suggested, and made the page URL the root page that has the search box and results. It appears to work in that it sends the query to the search bar; "https://mysite.sharepoint.com/site/SitePages/test.aspx?k=words" appears in the top bar. However, the search is not performed, even though I set the parameter name to k.

**Reply by gretchunkim** [reply]
@taliamath   Have you tried to use ?k as  the parameter instead?
